src/MapViewController.swift

```diff
@@ -417,11 +476,184 @@ open class MapViewController: UIViewController, LocationManagerDelegate, TGRecog
     tgViewController.markerRemove(marker)
     return
   }
+}
 
-  //MARK: - TGRecognizerDelegate
-
+extension MapViewController : TGMapViewDelegate, TGRecognizerDelegate {
+  
+  //MARK : TGMapViewDelegate
+  
+  open func mapView(_ mapView: TGMapViewController, didLoadSceneAsync scene: String) {
+    guard (loadDelegate != nil) && (self.loadDelegate?.responds(to: #selector(MapLoadDelegate.mapController(_:didLoadSceneAsync:))))! else {
```

Comment:
And actually, for some of these (this one in particular), since we're literally just calling through to the function, you could write it like

`loadDelegate?.mapController?(self, didLoadSceneAsync: scene)` - no guards or anything. Swift will handle all the checks for you.